django/db/models/options.py

```diff
@@ -992,6 +1021,16 @@ def _non_pk_concrete_field_names(self):
                     names.append(field.attname)
         return frozenset(names)
 
+    @cached_property
+    def _many_to_one_field_names(self):
+        """
+        Return a set of the many-to-one field names defined on the model.
+        """
+        names = []
+        for field in self.many_to_one:
+            names.append(field.name)
+        return frozenset(names)
```

Comment:
We could use a generator expression here:

```suggestion
        return frozenset(field.name for field in self.many_to_one)
```